src/game/GameWorldLoader.ts

```diff
@@ -0,0 +1,300 @@
+import {
+    GameWorld,
+    GameWorldModifier,
+    CardActionData,
+    EventCardActionData,
+    WorldQuery,
+} from './ContentTypes'
+import { Game, GameState, Card, CardAction, StateModifier, Stat } from './Types'
+import { Params, ParamQuery, hasMatchingParamQuery } from './Params'
+import { BasicGame } from './BasicGame'
+import { stateExtensionsFromData, createParameterCap } from './StateExtensions'
+
+/**
+ * Loads a GameWorld object as a Game<Params>.
+ * Uses the data structures to generate a runtime description of the game model.
+ *
+ * @param gameWorld The data definition of the game world
+ * @param random A function used for generating random values between 0 and 1
+ * @returns A Game<Params> object which uses the given data
+ */
+export function load(
+    gameWorld: GameWorld,
+    random: () => number = Math.random,
+): Game<Params> {
```

Comment:
Really nice work cleaning up the scenario loading process into a clear function!